### PR TITLE
[KOGITO-2015] - CLI: project creation: add `--enable-persistence` and…

### PR DESCRIPTION
**Pull request by Kaitou786**
… `--enable-events` to `--install-XXX` options

See: https://issues.redhat.com/browse/KOGITO-2015

Signed-off-by: TARUN KHANDELWAL <tarkhand@redhat.com>

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: cmd/kogito/command/install/kogito_operator.go [hunk 1]
```diff
@@ -79,6 +81,8 @@ func (i *installKogitoOperatorCommand) InitHook() {
 	i.command.Flags().BoolVar(&i.flags.installMgmtConsole, "install-mgmt-console", false, "Installs the default instance of Management Console being provisioned by the Kogito Operator in the project")
 	i.command.Flags().BoolVar(&i.flags.installAllServices, "install-all-services", false, "Installs the default instance of every Kogito Support services (Data Index, Jobs Service, etc.) being provisioned by the Kogito Operator in the project")
 	i.command.Flags().BoolVarP(&i.flags.force, "force", "f", false, "When set, the operator will be installed in the current namespace using a custom image, e.g. quay.io/kiegroup/kogito-cloud-operator:my-custom-tag")
+	i.command.Flags().BoolVar(&i.flags.enablePersistence, "enable-persistence", false, "If set to true will install Infinispan in the same namespace and inject the environment variables to configure the service connection to the Infinispan server.")
+	i.command.Flags().BoolVar(&i.flags.enableEvents, "enable-events", false, "If set to true will install a Kafka cluster via the Strimzi Operator. ")
```
Comment (radtriste) [inline cmd/kogito/command/install/kogito_operator.go:85]:
```suggestion
	i.command.Flags().BoolVar(&i.flags.enablePersistence, "enable-persistence", false, "If set will install Infinispan in the same namespace and inject the environment variables to configure the service connection to the Infinispan server.")
	i.command.Flags().BoolVar(&i.flags.enableEvents, "enable-events", false, "If set will install a Kafka cluster via the Strimzi Operator. ")
```

File: cmd/kogito/command/project/project.go [hunk 2]
```diff
@@ -35,6 +37,8 @@ func addProjectFlagsToCommand(command *cobra.Command, pFlags *projectFlags) {
 	command.Flags().BoolVar(&pFlags.installJobsService, "install-jobs-service", false, flags.InstallJobsService)
 	command.Flags().BoolVar(&pFlags.installManagementConsole, "install-mgmt-console", false, flags.InstallMgmtConsole)
 	command.Flags().BoolVar(&pFlags.installAll, "install-all", false, flags.InstallAllServices)
+	command.Flags().BoolVar(&pFlags.enablePersistence, "enable-persistence", false, "If set to true will install Infinispan in the same namespace and inject the environment variables to configure the service connection to the Infinispan server.")
+	command.Flags().BoolVar(&pFlags.enableEvents, "enable-events", false, "If set to true will install a Kafka cluster via the Strimzi Operator. ")
```
Comment (radtriste) [inline cmd/kogito/command/project/project.go:41]:
```suggestion
	command.Flags().BoolVar(&pFlags.enablePersistence, "enable-persistence", false, "If set will install Infinispan in the same namespace and inject the environment variables to configure the service connection to the Infinispan server.")
	command.Flags().BoolVar(&pFlags.enableEvents, "enable-events", false, "If set will install a Kafka cluster via the Strimzi Operator. ")
```

File: cmd/kogito/command/shared/create_default_services.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+package shared
+
+import (
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GetDefaultDataIndex gets the default Data Index instance
+func GetDefaultDataIndex(namespace string, enablePersistence bool, enableEvents bool) v1alpha1.KogitoDataIndex {
```
Comment (radtriste) [inline cmd/kogito/command/shared/create_default_services.go:24]: You cannot name the file `create_default_services.go` and the methods `GetDefault*`
Just call the file `default_services.go`

File: cmd/kogito/command/shared/default_services.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+package shared
+
+import (
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GetDefaultDataIndex gets the default Data Index instance
+func GetDefaultDataIndex(namespace string, enablePersistence bool, enableEvents bool) v1alpha1.KogitoDataIndex {
```
Comment (radtriste) [inline cmd/kogito/command/shared/default_services.go:24]: `enablePersistence bool, enableEvents bool` are useless here as data-index needs them to work

**Comment by sutaakar** [comment]
No, just checked the code here.

**Comment by radtriste** [comment]
me neither. let me check

**Comment by spolti** [comment]
@radtriste awesome, thanks.

Great work @Kaitou786.

**Comment by Kaitou786** [comment]
Thanks ! @spolti

**Pull request merged**